sdk/core/azure-core/tests/test_tracing_decorator.py

```diff
@@ -0,0 +1,181 @@
+            client = MockClient()
+            client.check_name_is_different()
+
+        assert len(parent.children) == 2
+        assert parent.children[0].name == "MockClient.__init__"
+        assert parent.children[1].name == "different name"
+
+    def test_used(self):
+        with FakeSpan(name="parent") as parent:
+            client = MockClient(policies=[])
+            client.get_foo(parent_span=parent)
+            client.get_foo()
+
+        assert len(parent.children) == 3
+        assert parent.children[0].name == "MockClient.__init__"
+        assert not parent.children[0].children
+        assert parent.children[1].name == "MockClient.get_foo"
+        assert not parent.children[1].children
+
+    def test_span_merge_span(self):
+        with FakeSpan(name="parent") as parent:
+            client = MockClient()
+            client.merge_span_method()
+            client.no_merge_span_method()
+
+        assert len(parent.children) == 3
+        assert parent.children[0].name == "MockClient.__init__"
+        assert not parent.children[0].children
+        assert parent.children[1].name == "MockClient.merge_span_method"
+        assert not parent.children[1].children
+        assert parent.children[2].name == "MockClient.no_merge_span_method"
+        assert parent.children[2].children[0].name == "MockClient.get_foo"
+
+    def test_span_complicated(self):
+        with FakeSpan(name="parent") as parent:
+            client = MockClient()
+            client.make_request(2)
+            with parent.span("child") as child:
+                time.sleep(0.001)
+                client.make_request(2, parent_span=parent)
+                assert FakeSpan.get_current_span() == child
+                client.make_request(2)
+
+        assert len(parent.children) == 4
+        assert parent.children[0].name == "MockClient.__init__"
+        assert not parent.children[0].children
+        assert parent.children[1].name == "MockClient.make_request"
+        assert not parent.children[1].children
+        assert parent.children[2].name == "child"
+        assert parent.children[2].children[0].name == "MockClient.make_request"
+        assert parent.children[3].name == "MockClient.make_request"
+        assert not parent.children[3].children
+
+    def test_span_with_exception(self):
+        """Assert that if an exception is raised, the next sibling method is actually a sibling span.
+        """
+        with FakeSpan(name="parent") as parent:
+            client = MockClient()
+            try:
+                client.raising_exception()
+            except:
+                pass
+            client.get_foo()
+
+        assert len(parent.children) == 3
+        assert parent.children[0].name == "MockClient.__init__"
+        assert parent.children[1].name == "MockClient.raising_exception"
+        # Exception should propagate status for Opencensus
+        assert parent.children[1].status == 'Something went horribly wrong here'
```

Comment:
Does `.status` have a value that can be maintained in successful cases?

Reply:
That's a good question, setting status is actually the resability of the plugin implementation (which are not in this PR). That being said, both OpenCensus and OpenTelemetry does nothing with status, until there is an exception when they make sure status is called. This test is mostly to check that it's possible to do that (the type of status in a real implementation doesn't have to be a string).
TLDR: I don't think it makes sense to check status on non-error scenario here (but we could in plugin implementation)